enhancements/insights/pulling-pull-secret-from-ocm.md

```diff
@@ -0,0 +1,168 @@
+and then updating the global `pull-secret` on the cluster with the new owner’s `pull-secret`.
+If the `pull-secret` is not changed on the cluster within 5 days, the ownership transfer is canceled
+and needs to be initiated again.
+This process applies to all scenarios that require cluster ownership transferred across Red Hat accounts.
+
+The intent is to simplify the cluster ownership process to a one step process.
+
+## Motivation
+
+The entry point to many OpenShift subscriptions is for a user to start with a 60 day trial from the OCM with personal credentials
+and then migrate it over to a company or organization account. In addition, reports from the field indicate that customers in general when transferring ownership, either follow one of the steps in the process and do not follow through with the other step required to complete the transfer. This enhancement will make the transfer frictionless and provide a better user experience.
+
+
+### Goals
+The goal of this enhancement is to describe the API and the interaction between OCM and the cluster that is being transferred to update the `pull-secret` on that cluster:
+
+- Extend the Insights Operator config with a new OCM API endpoint to be able to query the new `pull-secret` content
+- Periodically (once a day) query the new endpoint and if there is a new or updated content then update the `pull-secret`
+
+### Non-Goals
+- The enhancement will not describe the OCM user experience.
+- The enhancement will not provide a generic way to pull information from OCM into the cluster.
+- The process will not apply to disconnected clusters.
+
+## Proposal
+
+The Insights operator already communicates with the OCM API (see [Insights Operator pulling and exposing SCA certs from the OCM API](../insights/pulling-sca-certs-from-ocm.md). This is the proposed flow for updating the `pull-secret`:
+
+1. OCM provides an API that the cluster can call to check the current global `pull-secret`.
+2. User starts the migration process in the OCM by providing the `pull-secret` associated with the account to which the ownership will be transferred to.
+   * The details of whether the user provides the pull secret as a means of verification, or whether a request/approval workflow is implemented, is left to the OCM/UI flow. What is pertinent here is that there is an endpoint available to a cluster entity to fetch the new `pull-secret`.
+3. OCM validates the `pull-secret` for the new account and makes it available to the cluster for some time period i.e. the new API will now return the new `pull-secret` provided by the user.
+4. The Insights operator calls this API periodically, fetches the `pull-secret` and compares it with the current global `pull-secret` on the cluster.
+5. If there is a mismatch, the Insights operator applies the new `pull-secret`.
+   * Note: MCO updating the `pull-secret` on the nodes will not result in node reboots.
```

Comment:
Which is great, unless the pools are paused ;).  If this lands, we'll need doc updates [here][1], and maybe in other places, that say "if you want your cluster to survive a cred handoff, be very sure to unpause and level all of your nodes before you disable the outgoing secret" or some such.

[1]: https://docs.openshift.com/container-platform/4.9/updating/update-using-custom-machine-config-pools.html#update-using-custom-machine-config-pools-about_update-using-custom-machine-config-pools

Reply:
OK makes sense. Do we need to update the enhancement to highlight this? @mrunalp can you help herer please.

Reply:
I mentioned the docs update requirement

Reply:
are we confident in the docs, or do we want a robotic guard in the Insights operator that blocks on and complains about paused pools?  Not clear from this doc is how long the outgoing pull-secret will continue to work.  If we're just passing a cluster between users, sticking with the outgoing secret for a while doesn't seem like a big deal.  If we're rotating because someone leaked the old secret, your paused-pool nodes might be break.  Also not clear to me is how gracefully nodes recover from stale pull secrets; is there anything in reboot/pivot code that could lock up?  Maybe all the in-cluster logic should live in the MCO and be about waving alarms if there's a paused pool and a pending pull-secret change?  Thoughts?